src/Applicator/CouponCodeApplicator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Applicator;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Setono\SyliusGiftCardPlugin\Model\OrderInterface;
+use Sylius\Component\Core\Model\PromotionCouponInterface;
+use Sylius\Component\Order\Processor\OrderProcessorInterface;
+use Sylius\Component\Promotion\Action\PromotionApplicatorInterface;
+use Sylius\Component\Promotion\Repository\PromotionCouponRepositoryInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Webmozart\Assert\Assert;
+
+final class CouponCodeApplicator implements CouponCodeApplicatorInterface
```

Comment:
Doesn't Sylius have a class that does this?

Reply:
Sylius has applicators and processors, but none of them is flushing. So this applicator is just there to simplify the dependencies and to have a simpler code (controller especially)